### PR TITLE
feat: export `Watching` type

### PR DESCRIPTION
**Pull request by hardfist**
## Summary
export Watching and DevServe type for nx-labs, releated to #3861 

## Test Plan
No Test Needed

### CODE REVIEW

**Comment by h-a-n-a** [comment]
Can you merge it with regard to 0.3.0?

**Comment by hardfist** [comment]
> Can you merge it with regard to 0.3.0?

its no breaking so it could be merged to main

**Comment by h-a-n-a** [comment]
> > Can you merge it with regard to 0.3.0?
> 
> its no breaking so it could be merged to main

There's conflict with 0.3.0. You can postpone this rebasing operation to that branch for sure.

File: packages/rspack/src/index.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import type Watching from "./watching";
+import type { DevServer } from "./config";
```
Comment (h-a-n-a) [inline packages/rspack/src/index.ts:2]: This is not exported by webpack. Shouldn't we align this?

File: packages/rspack/src/index.ts [hunk 2]
```diff
@@ -1,3 +1,5 @@
+import type Watching from "./watching";
```
[the remark this answers is not in the document]
Comment (h-a-n-a) [reply packages/rspack/src/index.ts:1]: This is a definite task to do

File: packages/rspack/src/index.ts (hunk 1 above)
Comment (hardfist) [reply packages/rspack/src/index.ts:2]: yes，actually webpack can access this type by other way and they just dont export it at top level

**Comment by hardfist** [comment]
> > > Can you merge it with regard to 0.3.0?
> > 
> > 
> > its no breaking so it could be merged to main
> 
> There's conflict with 0.3.0. You can postpone this rebasing operation to that branch for sure.

we always need rebase before release so it seems ok for me

File: packages/rspack/src/index.ts (hunk 1 above)
Comment (h-a-n-a) [reply packages/rspack/src/index.ts:2]: Webpack does not directly export this, if you are trying to access the configuration of `devServer`, you can just follow `Configuration['devServer']`

Comment (hardfist) [reply packages/rspack/src/index.ts:2]: lets align with webpack first

Comment (h-a-n-a) [reply packages/rspack/src/index.ts:2]: Okay, let's remove `DevServer` then

**Review by h-a-n-a — APPROVED** [review]

**Review by h-a-n-a — APPROVED** [review]

**Pull request merged**